src/components/navigation-header/components/primary-nav-submenu/index.tsx

```diff
@@ -16,6 +16,9 @@ interface NavSubmenuProps {
 const PrimaryNavSubmenu = ({ ariaLabel, navItem }: NavSubmenuProps) => {
 	const { id, label } = navItem
 	const currentProduct = useCurrentProduct()
+	const navHeaderItems = currentProduct.navigationHeaderItems
+		? currentProduct.navigationHeaderItems[id] || []
+		: []
```

Comment:
💭 I don't think we want a default value here. If the code is erroring without the default value as a product is being onboarded, that's healthy friction.

Reply:
@ashleemboyer  That’s fair, thanks for calling this out!

This was initially just so I could focus on the problems I needed to solve, without having to write out all the nav data. I guess from my perspective there are other points of friction when trying to onboard new products, so I don’t know that “healthy” is the word I’d pick to describe the friction we’re creating. I’d personally prefer for this to be a warning in dev at least.

As well, in production, i’m feeling (maybe unwarranted?) a bit wary of intentionally throwing errors in render paths now that we’re in ISR-land. If the nav header fails to render for some weird unexpected reason, i think it’d be preferable to keep trying to render the rest of the experience, rather than bail completely.

Maybe this speaks to a broader need for a strategy for surfacing critical issues and gaps, but in a way that doesn’t totally prevent things from rendering 🤔